### DISCUSSION TITLE
Transposer is not working

### DISCUSSION THREAD

**Discussion by latimarkim** [tttapa/Control-Surface#786] [General]
Hello. I uploaded ‘transposer’ example to my arduino micro as it is but transposer didn’t work. However, every buttons for playing notes were working well. I tried ‘encoder selector’ with rotary encoder to transpose before but it sometimes worked, sometimes didn’t. My buttons including transposing ones were physically ok (I checked with my button checking code). I really wonder what is wrong. I put 1k resistors each to pull my buttons down, can it be a problem??

**Comment by tttapa** [comment]
The buttons should not be pulled down. You should connect each button between the appropriate GPIO pin and ground, without any external resistors, the internal pull-up resistor is enabled in software.

**Comment by latimarkim** [comment]
Thank you. I will remove resistor and try it again.

**Comment by latimarkim** [comment]
Thanks! The problem all solved!
